src/lib/services/api-token-service.limit.test.ts

```diff
@@ -0,0 +1,102 @@
+import { ApiTokenService } from './api-token-service';
+import { createTestConfig } from '../../test/config/test-config';
+import type { IUnleashConfig } from '../server-impl';
+import { ApiTokenType } from '../types/models/api-token';
+import FakeApiTokenStore from '../../test/fixtures/fake-api-token-store';
+import FakeEnvironmentStore from '../features/project-environments/fake-environment-store';
+import { createFakeEventsService } from '../../lib/features';
+import { ExceedsLimitError } from '../error/exceeds-limit-error';
+
+const createServiceWithLimit = (limit: number) => {
+    const config: IUnleashConfig = createTestConfig({
+        experimental: {
+            flags: {
+                resourceLimits: true,
+            },
+        },
+    });
+
+    const apiTokenStore = new FakeApiTokenStore();
+    const environmentStore = new FakeEnvironmentStore();
+    environmentStore.create({
+        name: 'production',
+        type: 'production',
+        enabled: true,
+        protected: true,
+        sortOrder: 1,
+    });
+
+    const eventService = createFakeEventsService(config);
+
+    const service = new ApiTokenService(
```

Comment:
it looks like you created an implicit `composition root` for the fake ApiTokenService. Since we touch this `not yet ported to the new architecture` code maybe it's worth creating a composition root for a real and fake API Token Service in the next PR?

Reply:
Aye, I could do that. Would you expect it to be in its own feature directory? And if so, would you expect the project-based API token controller and admin-based API token controller to both be there?

Reply:
You can check how we migrated nothing but the composition with createAccessService.ts. This way we don't need to do more than is needed to proceed.

Reply:
Addressed in https://github.com/Unleash/unleash/pull/7519